astro/src/content/docs/get-started/run-in-the-cloud/github-actions.mdx

```diff
@@ -0,0 +1,456 @@
+fa | 12:49:04.500 PM INFO JDBCMaintenanceModeDatabaseService - Latest Migration Version [1.48.1]
+fa | 12:49:04.500 PM INFO JDBCMaintenanceModeDatabaseService - Attempting to unlock database to support multi-node configurations
+fa | 12:49:04.501 PM INFO JDBCMaintenanceModeDatabaseService - Unlock completed
+
+```
+
+If you browse to FusionAuth again, you'll see that the user you created is still there.
+
+The FusionAuth application is separate from the FusionAuth database. In this case, the database is stored in a Docker volume. When the upgrade scripts run, the data should not be broken, and upgrading FusionAuth should be unnoticeable. However, mistakes can happen, as well as database crashes, and you should backup both the FusionAuth database and your application database daily. The rest of this article will show you how to automate login tests to check that your system still works after upgrading FusionAuth or your app. If something breaks, you can restore the old version of the database and begin debugging.
+
+For more details on upgrading FusionAuth, including non-silent upgrades, please read this [article](/docs/get-started/download-and-install/docker#upgrading).
+
+<Aside type="note">
+FusionAuth also has a paid version that [runs in the cloud](/docs/get-started/run-in-the-cloud/cloud). If you want to avoid the hassle of managing, backing up, and upgrading FusionAuth yourself, this might be the solution for you.
+</Aside>
+
+## Test Your App Login With Playwright
+
+FusionAuth is already running. To start the app that uses it, run the code below.
+
+```bash
+cd app
+npm install
+npm run start
+```
+
+Browse to `http://localhost:3000` and log in with the same user as previously.
+
+When changing your application, it would be faster to test that login works automatically, rather than by hand. This is called an integration, or end-to-end test, as opposed to a unit test, which tests only a single function. For integration tests in the browser, you can use Playwright.
+
+<Aside type="note">
+Playwright was created by Microsoft, runs in JavaScript, and allows you to run tests in multiple browsers. It is a popular successor to Selenium, which used Java, and Puppeteer, which was written by Google and focused on Chrome.
+</Aside>
+
+To start the Playwright tests for this app, open a new terminal and run the code below.
+
+```bash
+npx playwright install
+npx playwright install-deps
+npx playwright test --project=chromium;
+```
+
+The output should be as below.
+
+```bash
+Running 1 test using 1 worker
+
+  ✓  1 [chromium] › test.spec.js:3:1 › Test login (5.1s)
+
+  1 passed (5.8s)
+```
+
+The command returns `0` to the terminal if all tests passed, and another number if any tests failed. This allows you (and GitHub Actions) to write shell scripts that perform different tasks if the tests succeed or not.
+
+To see what is happening visually, run the code below.
+
+```bash
+npx playwright test --project=chromium --ui
+```
```

Comment:
I had to click the run button next to the Test to see it run.

Reply:
This image is not very readable.  May need to just take a section and zoom in so it is more clear.

![Screenshot 2024-02-22 at 4 03 12 PM](https://github.com/FusionAuth/fusionauth-site/assets/137117976/de474f7d-c16c-4b85-b537-78dd0fdd5c1f)